test/framework/clusterctl/repository.go

```diff
@@ -168,3 +170,35 @@ func YAMLForComponentSource(ctx context.Context, source ProviderVersionSource) (
 
 	return data, nil
 }
+
+// getComponentSourceFromURL fetches contents of component source YAML file from provided URL source.
+func getComponentSourceFromURL(source ProviderVersionSource) ([]byte, error) {
+	var buf []byte
+
+	u, err := url.Parse(source.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	switch u.Scheme {
```

Comment:
Will `u.Scheme` always be lower case? What happens if I gave the path as `HTTPS://github.com...`? Do we need to account for this?

If `url.Parse` handles this for use and lower cases them might be worth a quick note here for future readers

Reply:
Yep, `url.Parse` lower cases it: https://play.golang.org/p/TgabtcSjt7L. Will add comment here.